R/benchTradeStats.R

```diff
@@ -0,0 +1,296 @@
+#' Trade Execution Performance Benchmarks Statistical Testing
+#' 
+#' The main scope of the function is to gather statistical tests that are carried 
+#' on benchmarked trades performance, as obtained via \code{benchTradePerf()}. 
+#' When testing trading strategies on Symbols, assessing whether there is a statistical 
+#' significance difference in their performance is of interest. In other words,
+#' the goal is determining which given strategy outperformed the other or if they
+#' statistically bear the same results in terms of performance. All the statistical
+#' test included are \emph{non-parametric tests}, that is distribution-free tests.
+#' These tests allow great flexibility, but in turn require that data verifies 
+#' some assumptions in order for their results to be meaningful.  
+#' 
+#' There exists a wide range of algorithmic trading strategies and even whithin 
+#' the same category of strategies many sligthly different versions may exist 
+#' (they often differ by brokerage firm). In a post-trade analysis framework, 
+#' starting from trading strategies transactions prices we compare performance
+#' to test whether data supports a difference in median terms.
+#' 
+#' Two statistical testing approach are contemplated, the suitability of the 
+#' approach ultimately relies on analysts' specific research questions. In turn, 
+#' each of them critically depend on how transactional data was obtained in the 
+#' first place and the characteristics it has.
+#' First, the \emph{paired samples approach}, where: trades are two equal-length 
```

Comment:
Of course it would be in general, but this does not appear to be the case for the paired testing approach to be applicable. Here each order has to be, or have been, split into equal length pairs of child orders executed during the exact same timeframe.

Reply:
Overall timeframe must be equal agreed, but there is no requirement that the children be equal length or equal quantities. This will be impossible to constrain in production, and is irrelevant. I just want to be careful about the language when referring to the constraints for this test.

Reply:
Yes, after all we are considering the overall performance (or cost) metric of each child order only and therefore agree. However, I would recommend transactions happened under each of them to be very close in number, preferably approaching equality. Clearly, one may not guarantee such a circumstance in general. This leads to restrictions on the kinds of algorithmic trading strategies that can be suitably tested under this approach, namely the fact that it is mainly indicated as a testing framework for "static trade schedule algorithm(s)". These concepts are left out from function docs both because we are looking at the same references and because they may take rather long discussions. I have vignette discussion intentions in mind, where we can discuss them taking a wider perspective rather than strictly provide function usage in a general context. Your feedback on this documentations distinction would be much appreciated